src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Handlers/UserEventHandler.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.Extensions.Caching.Distributed;
+using OrchardCore.Users.Handlers;
+
+namespace OrchardCore.Users.TimeZone.Handlers;
+
+public class UserEventHandler : UserEventHandlerBase
+{
+    private const string CacheKey = "UserTimeZone/";
+
+    private readonly IDistributedCache _distributedCache;
+
+    public UserEventHandler(IDistributedCache distributedCache)
+    {
+        _distributedCache = distributedCache;
+    }
+
+    public override Task DeletedAsync(UserDeleteContext context)
+        => ForgetCacheAsync(context.User.UserName);
+
+    public override Task UpdatedAsync(UserUpdateContext context)
+        => ForgetCacheAsync(context.User.UserName);
+
+    public override Task DisabledAsync(UserContext context)
+        => ForgetCacheAsync(context.User.UserName);
+
+    private Task ForgetCacheAsync(string userName)
+    {
+        var key = GetCacheKey(userName);
+
+        return _distributedCache.RemoveAsync(key);
+    }
+
+    private static string GetCacheKey(string userName)
```

Comment:
I thinks it better to make this method `internal` instead of private. So it's called by `UserTimeZoneService`. This will tell us that these two classes need each other and the cache key is defined by another service.